src/elements/content-sidebar/activity-feed/task-new/Task.js

```diff
@@ -0,0 +1,166 @@
+    getMentionWithQuery?: Function,
+    getUserProfileUrl?: string => Promise<string>,
+    isDisabled?: boolean,
+    isPending?: boolean,
+    mentionSelectorContacts?: SelectorItems,
+    onAssignmentUpdate: Function,
+    onDelete?: Function,
+    onEdit?: Function,
+    translatedTaggedMessage?: string,
+    translations?: Translations,
+};
+
+const MAX_AVATARS = 3;
+
+// eslint-disable-next-line react/prefer-stateless-function
+class Task extends React.Component<Props> {
+    render(): React.Node {
+        const {
+            assigned_to,
+            created_at,
+            created_by,
+            currentUser,
+            due_at,
+            error,
+            id,
+            isPending,
+            onDelete,
+            onEdit,
+            onAssignmentUpdate = noop,
+            name,
+            status,
+            translatedTaggedMessage,
+            translations,
+            getAvatarUrl,
+            getUserProfileUrl,
+            getMentionWithQuery,
+            mentionSelectorContacts,
+        } = this.props;
+
+        // Enable when edit/delete are implemented for new tasks
+        const taskPermissions = {
+            can_edit: false,
+            can_delete: false,
+        };
+
+        // // find assignment for current user if permitted
+
+        const currentUserAssignment =
+            assigned_to && assigned_to.entries
+                ? assigned_to.entries.find(({ target }) => target.id === currentUser.id)
+                : null;
+
+        const assigneeCount = (assigned_to && assigned_to.entries.length) || 0;
+        const hiddenAssigneeCount = assigneeCount - MAX_AVATARS;
+        const isOverdue = due_at ? status === TASK_NEW_INCOMPLETE && new Date(due_at) < Date.now() : false;
+
+        return (
+            <div
+                className={classNames('bcs-task', {
+                    'bcs-is-pending': isPending || error,
+                })}
+                data-testid="task-card"
+            >
+                <Comment
+                    created_at={created_at}
+                    created_by={created_by.target}
+                    currentUser={currentUser}
+                    error={error}
+                    id={id}
+                    inlineDeleteMessage={messages.taskDeletePrompt}
+                    isPending={isPending}
+                    onDelete={onDelete}
+                    onEdit={onEdit}
+                    permissions={taskPermissions}
+                    tagged_message={name}
+                    translatedTaggedMessage={translatedTaggedMessage}
+                    translations={translations}
+                    getAvatarUrl={getAvatarUrl}
+                    getUserProfileUrl={getUserProfileUrl}
+                    mentionSelectorContacts={mentionSelectorContacts}
+                    getMentionWithQuery={getMentionWithQuery}
+                />
+                <div className="bcs-task-assignment-container">
+                    {due_at ? (
```

Comment:
Can this be coerced into a proper boolean so we can use `&&`?

Reply:
Yes, but I think the fact that && requires explicit coercion makes it less preferable than ternaries in general